cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp

```diff
@@ -110,45 +125,67 @@ class per_device_edgelist_t {
+        if (wgt)
+          wgt_copies.push_back(
+            std::make_tuple(wgt_->back().begin() + current_pos_, wgt->begin() + pos, copy_count));
+        if (edge_id)
+          edge_id_copies.push_back(std::make_tuple(
+            edge_id_->back().begin() + current_pos_, edge_id->begin() + pos, copy_count));
+        if (edge_type)
+          edge_type_copies.push_back(std::make_tuple(
+            edge_type_->back().begin() + current_pos_, edge_type->begin() + pos, copy_count));
+
+        count -= copy_count;
+        pos += copy_count;
+        current_pos_ += copy_count;
+      }
+    }
 
-    size_t count = src.size();
-    size_t pos   = 0;
+    std::for_each(src_copies.begin(), src_copies.end(), [&handle](auto tuple) {
+      raft::update_device(
+        std::get<0>(tuple), std::get<1>(tuple), std::get<2>(tuple), handle.get_stream());
+    });
+
+    std::for_each(dst_copies.begin(), dst_copies.end(), [&handle](auto tuple) {
+      raft::update_device(
+        std::get<0>(tuple), std::get<1>(tuple), std::get<2>(tuple), handle.get_stream());
+    });
 
-    while (count > 0) {
-      size_t copy_count = std::min(count, (src_.back().size() - current_pos_));
+    std::for_each(wgt_copies.begin(), wgt_copies.end(), [&handle](auto tuple) {
+      raft::update_device(
+        std::get<0>(tuple), std::get<1>(tuple), std::get<2>(tuple), handle.get_stream());
+    });
 
+    std::for_each(edge_id_copies.begin(), edge_id_copies.end(), [&handle](auto tuple) {
       raft::update_device(
-        src_.back().begin() + current_pos_, src.begin() + pos, copy_count, handle.get_stream());
+        std::get<0>(tuple), std::get<1>(tuple), std::get<2>(tuple), handle.get_stream());
+    });
+
+    std::for_each(edge_type_copies.begin(), edge_type_copies.end(), [&handle](auto tuple) {
       raft::update_device(
-        dst_.back().begin() + current_pos_, dst.begin() + pos, copy_count, handle.get_stream());
-      if (wgt)
-        raft::update_device(
-          wgt_->back().begin() + current_pos_, wgt->begin() + pos, copy_count, handle.get_stream());
-      if (edge_id)
-        raft::update_device(edge_id_->back().begin() + current_pos_,
-                            edge_id->begin() + pos,
-                            copy_count,
-                            handle.get_stream());
-      if (edge_type)
-        raft::update_device(edge_type_->back().begin() + current_pos_,
-                            edge_type->begin() + pos,
-                            copy_count,
-                            handle.get_stream());
-
-      count -= copy_count;
-      pos += copy_count;
-      current_pos_ += copy_count;
-
-      if (current_pos_ == src_.back().size()) { create_new_buffers(handle); }
-    }
+        std::get<0>(tuple), std::get<1>(tuple), std::get<2>(tuple), handle.get_stream());
+    });
 
-    handle.raft_handle().sync_stream();
+    handle.raft_handle().sync_stream(handle.get_stream());
```

Comment:
If we add get_stream() to mtmg::handle, what about adding sync_stream to mtmg::handle as well?